src/base/utils/misc.cpp

```diff
@@ -625,8 +627,16 @@ void Utils::Misc::openFolderSelect(const QString &absolutePath)
     if ((output == "dolphin.desktop") || (output == "org.kde.dolphin.desktop"))
         proc.startDetached("dolphin", QStringList() << "--select" << Utils::Fs::toNativePath(path));
     else if ((output == "nautilus.desktop") || (output == "org.gnome.Nautilus.desktop")
-             || (output == "nautilus-folder-handler.desktop"))
-        proc.startDetached("nautilus", QStringList() << "--no-desktop" << Utils::Fs::toNativePath(path));
+                 || (output == "nautilus-folder-handler.desktop")) {
+        proc.start("nautilus", {"--version"});
+        proc.waitForFinished();
+        const QString nautilusVerStr = QString(proc.readLine().simplified()).remove(QRegExp("[^0-9.]"));
+        using NautilusVersion = Utils::Version<int, 3>;
+        if (NautilusVersion::tryParse(nautilusVerStr, {1, 0, 0}) > NautilusVersion {3, 28})
+            proc.startDetached("nautilus", QStringList() << Utils::Fs::toNativePath(path));
```

Comment:
You missed my previous comment:
https://github.com/qbittorrent/qBittorrent/pull/8929#discussion_r188224674